src/Core/Workspace/Project.cs

```diff
@@ -143,6 +144,22 @@ public class Project
         internal bool UsesQuantumSdk =>
             Sdk.StartsWith("Microsoft.Quantum.Sdk");
 
+        private const string AutoLoadPropertyName = "IQSharpLoadAutomatically";
+        private bool? ShouldAutoLoad
+        {
+            get 
+            {
+                if (string.IsNullOrEmpty(ProjectFile)) return false;
```

Comment:
This, of course, expects that the `IQSharpLoadAutomatically` is defined in this file. MSBuild, though, supports importing props files and it's common practice to set values there. It would also be nice to be able to enable this property via include a nuget package.
Just highlighting that our current support for csproj files is very limited. Good enough for a v1 version but I truly hope we keep working on it.

Reply:
I totally agree. The real solution is https://github.com/microsoft/iqsharp/issues/277, to use MSBuild to understand the project semantics rather than parsing the file ourselves. Your earlier suggestion of using `dotnet build`, rather than encoding the build logic ourselves, seems to me like one of the best candidates, since it gets us support for all types of projects including Q# and C# projects (e.g., this would also give us support for projects that define custom magic commands or display encoders).